tests/unit/utils_memory.cpp

```diff
@@ -254,7 +254,7 @@ TEST(PoolResource, BigBlockAllocations) {
   TestMemory test_mem;
   const size_t max_blocks_per_chunk = 3U;
   const size_t max_block_size = 64U;
-  memgraph::utils::PoolResource mem(max_blocks_per_chunk, max_block_size, &test_mem);
+  memgraph::utils::PoolResource mem(max_blocks_per_chunk, max_block_size, &test_mem, &test_mem);
```

Comment:
I would use a different test memory resource to verify that big block are allocated with  the right resource.